### DISCUSSION TITLE
Installing python packages dependencies "requirements.txt"

### DISCUSSION THREAD

**Discussion by flachica** [Tecnativa/doodba#476] [Doodba - The template] [answered]
Could you tell me what commands I have to execute so that the dependencies are installed in the virtual environment used by the Odoo container?

If I do an `invoke img-build --pull` it tells me that the dependencies are installed. I restart Odoo but apparently the library is on an old version

This is cachetools and it gives me the error `name 'cachedmethod' is not defined` so I think what is happening is that the package is not in the proper virtual environment

**Comment by pedrobaeza** [comment, marked as answer]
Please read all the documentation before asking:

https://github.com/Tecnativa/doodba#optodoocustomdependenciestxt

**Reply by flachica** [reply]
Dear Pedro Baeza, I can assure you that I have read the documentation. After a few years in the sector I have learned that it is necessary. Which by the way, the documentation is very good, my congratulations, but I'm new to all this.

In the end I have stopped and started the container and the problem has been solved. It should be that my machine was not correctly recognizing the times that I restarted the container (without a **total stop**). I was going to consider the doubt resolved (as I did with [this one](https://github.com/Tecnativa/doodba/discussions/474)) but you have been faster than me.

It is very likely that I will ask newbie questions again, I even plan to collect these questions in an article to try to smooth the learning curve. We have all been through this at some point and this would be my way of saying thank you.

**Reply by pedrobaeza** [reply]
OK, I didn't want to be rush, but as you wrote the question, it was meant that the problem is not knowing where to put the dependencies. Next time, you can indicate in the original question the things you have already done, so no margin for the doubt. I recommend you this interesting reading: http://www.catb.org/~esr/faqs/smart-questions.html

Anyways, it seems your problems are not with Doodba itself, but maybe with Docker. A rule of thumb is to do `down` and then `up` instead of any other restarting mechanism to make sure everything is done again until you start to know which things require a cool restart or a hot one. It's the equivalent to "restart your computer" assessment.
